notifier/notifier.go

```diff
@@ -457,10 +457,10 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 
 	begin := time.Now()
 
-	// v1Payload and v2Payload represent 'alerts' marshaled for Alertmanager API
-	// v1 or v2. Marshaling happens below. Reference here is for caching between
+	// v2Payload represent 'alerts' marshaled for Alertmanager API v2.
+	// Marshaling happens below. Reference here is for caching between
 	// for loop iterations.
-	var v1Payload, v2Payload []byte
+	var v2Payload []byte
```

Comment:
I think we can just use `payload` here and simplify the logic, if in the future a v3 is added, we would have the possibility to think about another approach of supporting both versions.
(maybe a better approach than the logic we had here for v1/v2 (I can remember at least 3 bugs we had in here :)))

Reply:
I changed to cached payload. WDYT?